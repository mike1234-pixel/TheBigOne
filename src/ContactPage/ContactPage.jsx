import React, { useState } from "react";
import "./ContactPage.scss";

const ContactPage = () => {
  // state
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [email, setEmail] = useState("");
  const [message, setMessage] = useState("");

  const [errorMessage, setErrorMessage] = useState("");

  // handleSubmit
  const handleSubmit = (e) => {
    console.log("SUBMITTED");
    // send post request
    e.preventDefault();
  };

  // name handler
  const nameHandler = (e) => {
    setFirstName(e.target.value, validator(e));
  };

  // validate firstName
  const validator = (e) => {
    if (firstName !== "" && e.target.value.includes("1")) {
      setErrorMessage("Please enter a valid first Name");
    } else {
      setErrorMessage("");
    }
  };

  return (
    <div className="contact-page">
      <p>CONTACT ME PWEEEEASSSE X</p>
      <form onSubmit={() => handleSubmit()}>
        <input
          id="first-name-input"
          className="contact-input"
          value={firstName}
          onChange={(e) => nameHandler(e)}
        ></input>
        <input
          id="last-name-input"
          className="contact-input"
          value={lastName}
          onChange={(e) => setLastName(e.target.value)}
        ></input>
        <input
          id="email-input"
          className="contact-input"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
        ></input>
        <textarea
          id="message-input"
          className="contact-textarea"
          defaultValue="your message"
          value={message}
          onChange={(e) => setMessage(e.target.value)}
        ></textarea>
        <button type="submit">submit</button>
      </form>
      <p>{`firstName: ${firstName}`}</p>
      <p>{`lastName: ${lastName}`}</p>
      <p>{`email: ${email}`}</p>
      <p>{`message: ${message}`}</p>
      <p>{errorMessage}</p>
    </div>
  );
};

export default ContactPage;
