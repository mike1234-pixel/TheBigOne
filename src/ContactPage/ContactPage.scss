.dark-mode {
  color: #fff;
  background-image: url("../backgroundImages/backgroundImage1.webp");
  background-color: rgba(27, 27, 26, 0.5); // Tint color
  background-blend-mode: multiply;
}

.light-mode {
  color: #000;
  background-image: url("../backgroundImages/backgroundImage1.webp");
  background-color: rgba(211, 208, 206, 0.5); // Tint color
  background-blend-mode: multiply;
}

.contact-page {
  font-size: 20px;
  text-align: center;
  margin: 10px;
}

.contact-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  height: 70vh;
}

.contact-input,
.contact-textarea,
.contact-button {
  font-size: 30px;
  width: 30%;
  // margin: 10px;
}

.error-message {
  color: red;
}
