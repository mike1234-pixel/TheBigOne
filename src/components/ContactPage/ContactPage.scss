.contact-page {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 50px;
  min-height: 100vh;
  background-image: url("../HomePage/Carousel/carouselImages/img4.jpg");
  background-blend-mode: multiply;
  font-size: 20px;
  text-align: center;
  &-dark-mode {
    background-color: rgba(37, 37, 37, 0.5); // Tint color
    color: #fff;
  }
  &-light-mode {
    background-color: rgba(221, 221, 215, 0.5);
    color: #000;
  }
}

.contact-title {
  font-family: "Cabin Sketch";
  font-weight: 400;
  margin-bottom: 20px;
}

.contact-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  height: 70vh;
}

.contact-input,
.contact-textarea,
.contact-button {
  font-size: 30px;
  width: 100%;
  font-family: arial;
  // margin: 10px;
}

.contact-button {
  background: grey;
  color: #fff;
}

.contact-button::before {
  content: "";
  height: 50px;
  width: 50px;
  background: red;
}

.error-message {
  color: red;
}
