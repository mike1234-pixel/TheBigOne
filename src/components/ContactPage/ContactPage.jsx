import React, { useState } from "react";
import "./ContactPage.scss";
import axios from "axios";
import qs from "qs";

const ContactPage = (props) => {
  // state
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [email, setEmail] = useState("");
  const [message, setMessage] = useState("");

  const [firstNameErrorMessage, setFirstNameErrorMessage] = useState("");
  const [lastNameErrorMessage, setLastNameErrorMessage] = useState("");
  const [emailErrorMessage, setEmailErrorMessage] = useState("");
  const [messageError, setMessageError] = useState("");

  const [emailMessage, setEmailMessage] = useState(
    "We will never share your email address with anyone, ever."
  );

  // handleSubmit
  const handleSubmit = (e) => {
    e.preventDefault();
    if (firstName === "" || lastName === "" || email === "" || message === "") {
      setEmailMessage(
        "Please enter a first name, last name, email and message."
      );
      return;
    }
    let payload = {
      firstName: firstName,
      lastName: lastName,
      email: email,
      message: message,
    };

    axios
      .post("http://localhost:4000/contact", qs.stringify(payload))
      .then((err, res) => {
        if (err) {
          console.log(err);
        } else {
          console.log(res);
        }
      });

    setEmailMessage("Thanks for your message. We will be in touch shortly.");
    setFirstName("");
    setLastName("");
    setEmail("");
    setMessage("");
  };

  // ONCHANGE HANDLERS
  const firstNameHandler = (e) => {
    setFirstName(e.target.value, firstNameValidator(e));
  };

  const lastNameHandler = (e) => {
    setLastName(e.target.value, lastNameValidator(e));
  };

  const emailHandler = (e) => {
    setEmail(e.target.value, emailValidator(e));
  };

  const messageHandler = (e) => {
    setMessage(e.target.value, messageValidator(e));
  };

  // VALIDATORS
  // error message appears if: name has chars other than letters or length greater than 20 or it contains a space
  const firstNameValidator = (e) => {
    if (
      (e.target.value !== "" && /^[a-zA-Z]+$/.test(e.target.value) === false) ||
      e.target.value.length > 20 ||
      /␣/g.test(e.target.value)
    ) {
      setFirstNameErrorMessage("Please enter a valid first name");
    } else {
      setFirstNameErrorMessage("");
    }
  };

  const lastNameValidator = (e) => {
    if (
      (e.target.value !== "" && /^[a-zA-Z]+$/.test(e.target.value) === false) ||
      e.target.value.length > 20 ||
      /␣/g.test(e.target.value)
    ) {
      setLastNameErrorMessage("Please enter a valid last name name");
    } else {
      setLastNameErrorMessage("");
    }
  };

  // email must contain at least one @ and one .
  const emailValidator = (e) => {
    if (
      e.target.value !== "" &&
      /[@]/.test(e.target.value) === false &&
      /[\.]/.test(e.target.value) === false
    ) {
      setEmailErrorMessage("Please enter a valid email");
    } else {
      setEmailErrorMessage("");
    }
  };

  // message must be 10 characters or longer
  const messageValidator = (e) => {
    if (e.target.value.length < 10) {
      setMessageError("Message must 10 characters or longer.");
    } else {
      setMessageError("");
    }
  };

  return (
    <div className="contact-page">
      <h1 className="contact-title">Contact Me</h1>
      <form
        className={
          props.darkMode ? `dark-mode contact-form` : `light-mode contact-form`
        }
        onSubmit={(e) => handleSubmit(e)}
      >
        <input
          id="first-name-input"
          className="contact-input"
          value={firstName}
          placeholder="First Name"
          onChange={(e) => firstNameHandler(e)}
        ></input>
        <p className="error-message">{firstNameErrorMessage}</p>
        <input
          id="last-name-input"
          className="contact-input"
          value={lastName}
          placeholder="Last Name"
          onChange={(e) => lastNameHandler(e)}
        ></input>
        <p className="error-message">{lastNameErrorMessage}</p>
        <input
          id="email-input"
          className="contact-input"
          value={email}
          placeholder="Email"
          onChange={(e) => emailHandler(e)}
        ></input>
        <p className="error-message">{emailErrorMessage}</p>
        <textarea
          id="message-input"
          className="contact-input"
          placeholder="Your Message"
          value={message}
          onChange={(e) => messageHandler(e)}
        ></textarea>
        <p className="error-message">{messageError}</p>
        <p>{emailMessage}</p>
        <button type="submit" className="contact-button">
          submit
        </button>
      </form>
    </div>
  );
};

export default ContactPage;
