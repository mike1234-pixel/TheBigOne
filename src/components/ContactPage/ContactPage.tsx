import React, { useState, useEffect } from "react";
import "./ContactPage.scss";
import axios from "axios";
import qs from "qs";
import PropTypes from "prop-types";
import ReCAPTCHA from "react-google-recaptcha"; // https://github.com/dozoisch/react-google-recaptcha

const ContactPage = (props) => {
  // state
  const [firstName, setFirstName] = useState<string>("");
  const [lastName, setLastName] = useState<string>("");
  const [email, setEmail] = useState<string>("");
  const [message, setMessage] = useState<string>("");

  const [firstNameErrorMessage, setFirstNameErrorMessage] = useState<string>(
    ""
  );
  const [lastNameErrorMessage, setLastNameErrorMessage] = useState<string>("");
  const [emailErrorMessage, setEmailErrorMessage] = useState<string>("");
  const [messageError, setMessageError] = useState<string>("");

  const [emailMessage, setEmailMessage] = useState<string>(
    "We will never share your email address with anyone, ever."
  );

  // payload interface [TS]
  interface contactFormMessagePayload {
    firstName: String;
    lastName: String;
    email: String;
    message: String;
  }

  // handleSubmit
  const handleSubmit = (e): void => {
    e.preventDefault();
    if (firstName === "" || lastName === "" || email === "" || message === "") {
      setEmailMessage(
        "Please enter a first name, last name, email and message."
      );
      return;
    }
    let payload: contactFormMessagePayload = {
      firstName: firstName,
      lastName: lastName,
      email: email,
      message: message,
    };

    axios
      .post("http://localhost:4000/contact", qs.stringify(payload))
      .then((err) => {
        if (err) {
          console.log(err);
        }
      });

    setEmailMessage("Thanks for your message. We will be in touch shortly.");
    setFirstName("");
    setLastName("");
    setEmail("");
    setMessage("");
  };

  // ONCHANGE HANDLERS
  const firstNameHandler = (e): void => {
    setFirstName(e.target.value);
    firstNameValidator(e);
  };

  const lastNameHandler = (e): void => {
    setLastName(e.target.value);
    lastNameValidator(e);
  };

  const emailHandler = (e): void => {
    setEmail(e.target.value);
    emailValidator(e);
  };

  const messageHandler = (e): void => {
    setMessage(e.target.value);
    messageValidator(e);
  };

  // VALIDATORS
  // error message appears if: name has chars other than letters or length greater than 20 or it contains a space
  const nameValidator = (name: string): boolean => {
    if (
      (name !== "" && /^[a-zA-Z]+$/.test(name) === false) ||
      name.length > 20 ||
      /␣/g.test(name)
    ) {
      return true;
    }
    return false;
  };

  const firstNameValidator = (e): void => {
    if (nameValidator(e.target.value)) {
      setFirstNameErrorMessage("Please enter a valid first name");
    } else {
      setFirstNameErrorMessage("");
    }
  };

  const lastNameValidator = (e): void => {
    if (nameValidator(e.target.value)) {
      setLastNameErrorMessage("Please enter a valid last name name");
    } else {
      setLastNameErrorMessage("");
    }
  };

  // email must contain at least one @ and one .
  const emailValidator = (e): void => {
    if (e.target.value !== "" && /[@]/.test(e.target.value) === false) {
      setEmailErrorMessage("Please enter a valid email");
    } else {
      setEmailErrorMessage("");
    }
  };

  // message must be 10 characters or longer
  const messageValidator = (e): void => {
    if (e.target.value.length < 10) {
      setMessageError("Message must 10 characters or longer.");
    } else {
      setMessageError("");
    }
  };

  return (
    <div
      className={
        props.darkMode ? `dark-mode contact-page` : `light-mode contact-page`
      }
    >
      <h1 className="contact-title">Contact Me</h1>
      <form className="contact-form" onSubmit={(e) => handleSubmit(e)}>
        <input
          id="first-name-input"
          className="contact-input"
          value={firstName}
          placeholder="First Name"
          onChange={(e) => firstNameHandler(e)}
        ></input>
        <p className="error-message">{firstNameErrorMessage}</p>
        <input
          id="last-name-input"
          className="contact-input"
          value={lastName}
          placeholder="Last Name"
          onChange={(e) => lastNameHandler(e)}
        ></input>
        <p className="error-message">{lastNameErrorMessage}</p>
        <input
          id="email-input"
          className="contact-input"
          value={email}
          placeholder="Email"
          onChange={(e) => emailHandler(e)}
        ></input>
        <p className="error-message">{emailErrorMessage}</p>
        <textarea
          id="message-input"
          className="contact-input"
          placeholder="Your Message"
          value={message}
          onChange={(e) => messageHandler(e)}
        ></textarea>
        <p className="error-message">{messageError}</p>
        <p>{emailMessage}</p>
        <button type="submit" className="contact-button">
          submit
        </button>
        <ReCAPTCHA sitekey="site key" onChange={(e) => handleSubmit(e)} />
      </form>
    </div>
  );
};

export default ContactPage;

ContactPage.propTypes = {
  darkMode: PropTypes.bool,
};
