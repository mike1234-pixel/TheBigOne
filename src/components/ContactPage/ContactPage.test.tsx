import React from "react";
import ContactPage from "./ContactPage";
import renderer from "react-test-renderer";

// snapshot test

it("renders correctly", () => {
  const tree = renderer.create(<ContactPage darkMode={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
