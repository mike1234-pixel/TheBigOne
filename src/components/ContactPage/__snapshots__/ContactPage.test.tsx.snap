// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="contact-page"
>
  <h1
    className="contact-title"
  >
    Contact Me
  </h1>
  <form
    className="dark-mode contact-form"
    onSubmit={[Function]}
  >
    <input
      className="contact-input"
      id="first-name-input"
      onChange={[Function]}
      placeholder="First Name"
      value=""
    />
    <p
      className="error-message"
    >
      
    </p>
    <input
      className="contact-input"
      id="last-name-input"
      onChange={[Function]}
      placeholder="Last Name"
      value=""
    />
    <p
      className="error-message"
    >
      
    </p>
    <input
      className="contact-input"
      id="email-input"
      onChange={[Function]}
      placeholder="Email"
      value=""
    />
    <p
      className="error-message"
    >
      
    </p>
    <textarea
      className="contact-input"
      id="message-input"
      onChange={[Function]}
      placeholder="Your Message"
      value=""
    />
    <p
      className="error-message"
    >
      
    </p>
    <p>
      We will never share your email address with anyone, ever.
    </p>
    <button
      className="contact-button"
      type="submit"
    >
      submit
    </button>
  </form>
</div>
`;
